moveit_ros/occupancy_map_monitor/include/moveit/occupancy_map_monitor/occupancy_map_monitor.h

```diff
@@ -137,26 +244,22 @@ class OccupancyMapMonitor
   bool getShapeTransformCache(std::size_t index, const std::string& target_frame, const rclcpp::Time& target_time,
                               ShapeTransformCache& cache) const;
 
+  std::unique_ptr<MiddlewareHandle> middleware_handle_;
   std::shared_ptr<tf2_ros::Buffer> tf_buffer_;
   std::string map_frame_;
   double map_resolution_;
-  boost::mutex parameters_lock_;
+  std::mutex parameters_lock_;
```

Comment:
This could be marked `mutable`.

Reply:
I'm not sure this mutex is even needed (if you look at its use it seems silly).  However, I was trying to not include a refactor of that in this change as I want to refactor all the parameter stuff at once.

Instead, I am just doing a basic boost->stl refactor along with these changes where it makes sense.  I can make this mutable but that wouldn't matter much.